src/utils/upload/processUploadV1.js

```diff
@@ -0,0 +1,272 @@
+        },
+      ),
+    );
+
+    return;
+  }
+
+  dispatch(
+    updateSampleFile(
+      sampleUuid,
+      fileName,
+      {
+        ...filePropertiesToUpdate,
+        upload: {
+          status: UploadStatus.UPLOADED,
+          progress: 100,
+          amplifyPromise: null,
+        },
+      },
+    ),
+  );
+};
+
+const renameFileIfNeeded = (fileName, type) => {
+  // rename files to include .gz
+  const uncompressed = !['application/gzip', 'application/x-gzip'].includes(type) && !fileName.endsWith('.gz');
+  let newFileName = uncompressed ? `${fileName}.gz` : fileName;
+
+  // We rename genes.tsv files to features.tsv (for a single entry)
+  newFileName = newFileName.replace('genes', 'features');
+
+  return newFileName;
+};
+
+const uploadSingleFile = (newFile, activeProjectUuid, sampleUuid, dispatch) => {
+  const metadata = metadataForFile(newFile);
+
+  const newFileName = renameFileIfNeeded(newFile.fileObject.name, newFile.fileObject.type);
+
+  compressAndUploadSingleFile(
+    activeProjectUuid, sampleUuid, newFileName, newFile, dispatch, metadata,
+  );
+
+  return [newFileName, { ...newFile, name: newFileName }];
+};
+
+const compressAndUpload = (sample, activeProjectUuid, dispatch) => Object.fromEntries(
+  Object.values(sample.files)
+    .map((file) => uploadSingleFile(file, activeProjectUuid, sample.uuid, dispatch)),
+);
+
+const processUpload = async (filesList, sampleType, samples, activeProjectUuid, dispatch) => {
+  const samplesMap = filesList.reduce((acc, file) => {
+    const pathToArray = file.name.trim().replace(/[\s]{2,}/ig, ' ').split('/');
+
+    const sampleName = pathToArray[0];
+    const fileName = _.last(pathToArray);
+
+    // Update the file name so that instead of being saved as
+    // e.g. WT13/matrix.tsv.gz, we save it as matrix.tsv.gz
+    file.name = fileName;
+
+    const sampleUuid = Object.values(samples).filter(
+      (s) => s.name === sampleName
+        && s.projectUuid === activeProjectUuid,
+    )[0]?.uuid;
+
+    return {
+      ...acc,
+      [sampleName]: {
+        ...acc[sampleName],
+        uuid: sampleUuid,
+        files: {
+          ...acc[sampleName]?.files,
+          [fileName]: file,
+        },
+      },
+    };
+  }, {});
+
+  Object.entries(samplesMap).forEach(async ([name, sample]) => {
+    // Create sample if not exists.
+    try {
+      sample.uuid ??= await dispatch(createSample(activeProjectUuid, name, sampleType));
+    } catch (e) {
+      // If sample creation fails, sample should not be created
+      return;
+    }
+
+    sample.files = compressAndUpload(sample, activeProjectUuid, dispatch);
+
+    Object.values(sample.files).forEach((file) => {
+      // Create files
+      dispatch(
+        updateSampleFile(
+          sample.uuid,
+          file.name,
+          {
+            ...file,
+            path: `${activeProjectUuid}/${file.name.replace(name, sample.uuid)}`,
+          },
+        ),
+      );
+    });
+  });
+};
```

Comment:
why is this piece of code no longer needed for v2?

Reply:
It's to create the files. It is needed, but now we call one action in particualr for creating the sample files isntead of reusing `updateSampleFile` (we call `createSampleFileV2`)